tensorflow_federated/python/learning/keras_utils.py

```diff
@@ -111,14 +123,49 @@ def from_keras_model(keras_model,
     ValueError: If `keras_model` was compiled.
   """
   py_typecheck.check_type(keras_model, tf.keras.Model)
+  py_typecheck.check_type(loss, (tf.keras.losses.Loss,
+                                 collections.Sequence,
+                                 collections.Mapping))
+
+  if loss_weights is not None:
+    py_typecheck.check_type(loss, (collections.Sequence,
+                                   collections.Mapping))
+
+  if isinstance(loss, collections.Sequence):
+    if len(loss) != len(keras_model.outputs):
+      raise ValueError('`keras_model` must have equal number of '
+                       'outputs and losses')
+
+    if loss_weights is not None and len(loss) != len(loss_weights):
+      raise ValueError('`keras_model` must have equal number of '
+                       'losses and loss weights')
+
   if keras_model._is_compiled:  # pylint: disable=protected-access
     raise ValueError('`keras_model` must not be compiled. Use '
                      'from_compiled_keras_model() instead.')
+
   dummy_tensors = _preprocess_dummy_batch(dummy_batch)
   if optimizer is None:
+    if isinstance(loss, collections.Mapping):
+      loss_functions = []
+      for name in keras_model.output_names:
+        if name not in loss:
+          logging.warning(
```

Comment:
TFF tries to be strict and avoid assumptions on user input. Lets change this to raise an error instead, and include the names of the outputs and losses so the user can easily determine what was missing.